Fix catalog page graph browser back navigation issue

mahendra1290 opened the pull request:
## Hey, I just made a Pull Request!

Fixed the navigation issue on the catalog graph page, where the navigation history was not correctly maintained when users clicked on nodes and then used the browser's back button.

closes: #25890

### Before

https://github.com/user-attachments/assets/4503dbf2-d7f0-4e91-b62d-31b2344bb329

### After

https://github.com/user-attachments/assets/575434df-6518-462e-9503-976c9bf66700

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

benjdlambert commented:
Hey :wave: Thanks for this, looks like there's some tests failing, most likely due to some of the mocks not having the required methods. You'll also need a changeset, which you can create by running `yarn changeset`. Some [more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets) on changesets too.

mahendra1290 commented:
> Hey 👋 Thanks for this, looks like there's some tests failing, most likely due to some of the mocks not having the required methods. You'll also need a changeset, which you can create by running `yarn changeset`. Some [more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets) on changesets too.

I have fixed and improved the tests which now tests browser navigation also, and also added changesets please have a look and provide your feedback

mahendra1290 commented:
@benjdlambert please review this whenever you get time 😃

benjdlambert reviewed (approved):
Nice thanks! This looks much simpler! :tada:
